tests, net: Convert slirp tests to use the binding plugin directly

EdDev opened the pull request:
### What this PR does

Using the binding plugin directly avoids the need to access an external registry (quay.io) to fetch the default sidecar image.

The test now uses the built sidecar from the sources.

<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: -->
Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
```release-note
NONE
```

EdDev commented:
/test pull-kubevirt-e2e-k8s-1.29-sig-network

EdDev commented:
/cc @ormergi

EdDev commented on tests/network/vmi_slirp_interface.go:
```diff
@@ -46,18 +49,34 @@ import (
+			libvmi.WithInterface(v1.Interface{
+				Name:       v1.DefaultPodNetwork().Name,
+				Binding:    &v1.PluginBinding{Name: slirpBindingName},
+				MacAddress: "de:ad:00:00:be:af",
```
Yes it was, on line 60.
> vmi.Spec.Domain.Devices.Interfaces[0].MacAddress = "de:ad:00:00:be:af"

AlonaKaplan commented on tests/network/vmi_slirp_interface.go:
```diff
@@ -46,18 +49,34 @@ import (
 )
 
-var _ = SIGDescribe("Slirp", decorators.Networking, func() {
+var _ = SIGDescribe("Slirp", decorators.Networking, Serial, func() {
```
decorators.NetCustomBindingPlugins

AlonaKaplan commented on tests/network/vmi_slirp_interface.go:
```diff
@@ -23,6 +23,9 @@ import (
 	"context"
```
For consistency, I would change the file name to `bindingplugin_slirp`.

EdDev commented:
[change](https://github.com/kubevirt/kubevirt/compare/bd6b7e2994a33687390012e98fe9ac1a6f38ca7a..33876de31027ca25d82c6a283dcf528b3d02544e): Renamed test file and added a label to the test per the review.
(The rename is added as a separate commit to help with the review)

AlonaKaplan commented:
/approve
